.github/workflows/verify-gha-unit-tests-count.yml

```diff
@@ -50,7 +50,7 @@ jobs:
           echo All root unit tests count: ${{ env.root_all_unit_tests_count }}
           echo All shards unit tests count: ${{ env.shards_all_unit_tests_count }}
 
-      - name: verify unit tests count
+      - name: fail the check
```

Comment:
This step doesn't always fail the check.  "verify" was a better verb for the name than "fail".

Reply:
I updated the name because this step only runs when there's a difference in the number of tests